src/v/cluster_link/source_topic_syncer.cc

```diff
@@ -143,123 +252,300 @@ ss::future<> source_topic_syncer::run_impl() {
-            .source_topic_name = source_topic_name,
-            .destination_topic_id = ::model::topic_id{::uuid_t::create()},
-            .partition_count = it->second.partition_count,
-            .replication_factor = it->second.rf,
-            .topic_configs = std::move(configs)}});
+
+        // TODO: Once Topic IDs are supported, check that the Topic ID in the
+        // metadata response matches the expecteed Topic ID for this topic.
+
+        // If we detect that the partition count has gone down, this indicates
+        // that the topic may have been deleted and then re-created, so we will
+        // put the topic into the failed state
+        if (
+          mirror_topic_cache.partition_count > metadata_cache.partition_count) {
+            vlog(
+              logger().warn,
+              "Topic {} has fewer partitions than expected, marking as failed",
+              topic);
+            commands.emplace_back(model::update_mirror_topic_state_cmd{
+              .topic = topic,
+              .state = model::mirror_topic_state::failed,
+            });
+            continue;
+        }
+
+        // Detect if the partition count has changed
+        if (
+          mirror_topic_cache.partition_count
+          != metadata_cache.partition_count) {
+            vlog(
+              logger().trace,
+              "Topic {} has updated its partition count: {} -> {}",
+              topic,
+              mirror_topic_cache.partition_count,
+              metadata_cache.partition_count);
+            enqueue_command = true;
+        }
+
+        // Detect if RF has changed
+        if (mirror_topic_cache.replication_factor != metadata_cache.rf) {
+            vlog(
+              logger().trace,
+              "Topic {} has updated its RF: {} -> {}",
+              topic,
+              mirror_topic_cache.replication_factor,
+              metadata_cache.rf);
+            enqueue_command = true;
+        }
+
+        auto configs = validate_and_get_configs_from_response(
+          logger(), describe_result);
+        if (!configs.has_value()) {
+            vlog(
+              logger().trace,
+              "Failed to validate describe result for topic {}",
+              describe_result.resource_name);
+            // If RF or partition count changed, then we will use cached values
+            // for that topics configs
+            if (enqueue_command) {
+                vlog(
+                  logger().trace,
+                  "Metadata updated for topic {}.  Using cached property "
+                  "values",
+                  topic);
+                commands.emplace_back(model::update_mirror_topic_properties_cmd{
+                  .topic = topic,
+                  .partition_count = metadata_cache.partition_count,
+                  .replication_factor = metadata_cache.rf,
+                  .topic_configs = copy_hash_map(
+                    mirror_topic_cache.topic_configs),
+                });
```

Comment:
nit: consider storing partition_count and replication factor in a variable and just override it here. Then we will not need two separate emplace back